### PR TITLE
Solution: Make Debug Logging Configurable

### PR DESCRIPTION
**Pull request by DerTim1**
_See also #331_

With this new configuration_option `debug_logging: false` (defaults to `true`) you can mute the Quantum debug-messages (if you want to use the `debug`-level of Logger for your own stuff).

### CODE REVIEW

**Comment by maennchen** [comment]
@DerTim1 Would you add one tests to cover if the code actually works? (I don't think that it is not working but would like to avoid regressions.)

File: lib/quantum/job_broadcaster.ex [hunk 1]
```diff
@@ -18,24 +18,25 @@ defmodule Quantum.JobBroadcaster do
...
+  @spec start_link(GenServer.server(), [Job.t()], boolean) :: GenServer.on_start()
+  def start_link(name, jobs, debug_logging) do
     __MODULE__
-    |> GenStage.start_link(jobs, name: name)
+    |> GenStage.start_link({jobs, debug_logging}, name: name)
     |> Util.start_or_link()
   end
 
   @doc false
   @spec child_spec({GenServer.server(), [Job.t()]}) :: Supervisor.child_spec()
```
Comment (maennchen) [inline lib/quantum/job_broadcaster.ex]: You need to add `boolean` to this type spec.

`@spec child_spec({GenServer.server(), [Job.t()], boolean()}) :: Supervisor.child_spec()`

This is also the reason why the tests do not pass.

**Comment by DerTim1** [comment]
@maennchen Thank you! 
I fixed the mistake with the spec and also added one test for capture or not capture the debug logging.

**Comment by obrok** [comment]
@maennchen Could you please release this?

**Comment by florinpatrascu** [comment]
I second @maennchen 's comment. Is the debug logging configuration feature that keeps us from using Quantum from hex.pm. Is there an ETA for publishing a new release containing this feature? Thank you.

**Pull request merged**